### ISSUE TITLE
unable to inherit credentials for the child account version 5.1.0

### ISSUE THREAD

**Issue by Amiralizim** [99designs/aws-vault#489]
for version 5.1.0 my config file looks something like this: 
[profile source]
output=json
region=us-east-2
mfa_serial=arn:aws:iam::123456789123:mfa/<profileName>

[profile dev]
source_profile=source
output=json
region=us-east-2
mfa_serial=arn:aws:iam::12345678912:mfa/<profileName>
role_arn=arn:aws:iam::123457891:role/admin

after adding the profile the credentials doesn't get added for dev proile for some reason so I'm unable to aws-vault exec --no-session --duration=3h dev -- env for dev profile as a result

**Comment by mbelang** [comment]
Yeah same here we 2 of my colleagues that have updated to 5.1.1 and generating creds from source profile doesn't work anymore.

**Comment by stephencoe** [comment]
Same, the workaround described in https://github.com/99designs/aws-vault/issues/381#issuecomment-501637035 works for me to get this working. no source profile, adding credentials for each profile

## Not working (example from the docs)
```
[profile work]
mfa_serial=arn:aws:iam::11111111111:mfa/my_user

[profile read_only]
role_arn=arn:aws:iam::222222222222:role/read_only
source_profile=work
region=eu-west-1

[profile prod]
role_arn=arn:aws:iam::3333333333333:role/read_only
source_profile=work
region=eu-west-1

[profile admin]
role_arn=arn:aws:iam::222222222222:role/admin
source_profile=work
region=eu-west-1
```
Error: `aws-vault: error: exec: Failed to get credentials for admin: AccessDenied: User: arn:aws:iam::1111111111111:user/my_user is not authorized to perform: sts:AssumeRole on resource: arn:aws:iam::2222222222222:role/admin`

## Works (adding credentials manually per profile)
```
[profile work]
mfa_serial=arn:aws:iam::11111111111:mfa/my_user

[profile read_only]
role_arn=arn:aws:iam::222222222222:role/read_only
mfa_serial=arn:aws:iam::11111111111:mfa/my_user
region=eu-west-1

[profile prod]
mfa_serial=arn:aws:iam::11111111111:mfa/my_user
role_arn=arn:aws:iam::3333333333333:role/read_only
region=eu-west-1

[profile admin]
mfa_serial=arn:aws:iam::11111111111:mfa/my_user
role_arn=arn:aws:iam::222222222222:role/admin
region=eu-west-1
```

**Comment by mtibben** [comment]
I believe this should be resolved in v5.1.2. Re-open if not
